static/js/settings_user_groups.js

```diff
@@ -72,7 +77,16 @@ export function populate_user_groups() {
         const userg = $(`div.user-group[id="${CSS.escape(data.id)}"]`);
         for (const user_id of data.members) {
             const user = people.get_by_user_id(user_id);
-            user_pill.append_user(user, pills);
+            // we're using `.call` to bind `user_pill_config` to `this`, yuck!
+            // This is one way to make `user_pill_config` available to functions
+            // that send values to input_pills... we use `user_pill_config` to
+            // have settings_user_groups not show user status emoji, but allow
+            // the compose and typeahead to. (Just having a variable in
+            // user_pills wouldn't work because it would be shared across
+            // containers and break.) An alternative approach would be to use a
+            // function generator, but the best solution would probably be to
+            // redo the pills system using modern JavaScript classes. TODO?
+            user_pill.append_user.call(user_pill_config, user, pills);
```

Comment:
This is pretty ugly and I think we want a better solution.  Why can't we just add `user_pill_config` as a parameter on the existing pill container objects when they are initialized?

Reply:
Right, I was trying to prevent `emoji_status_info` from being passed down to input_pills so that it's not rendered into the template. Except it's simpler to just add `pill_config` to the `input_pill` `store` and check for that in `input_pill`.

I've updated to that approach.

Reply:
lgtm.